### PR TITLE
#13052: Adding Tests for Printing Destination Registers and fix `dprint_tensix_dest_reg` for Float32 Format

### PR DESCRIPTION
**Pull request by miacim**
### Ticket
#13052 - Add tests to validate printing of destination registers.

### Problem Description

The new Blackhole feature in #12234 requires additional test infrastructure to validate the printing functionality of destination registers. Currently, there are no tests that ensure `dprint_tensix_dest_reg` correctly handles various formats.

### Changes Made

- Updated `dprint_tensix_dest_reg` to support printing destination registers in the `float32` format.
- Added two unit tests to validate the printing of destination registers:
  - One test for a 16-bit destination register.
  - One test for a 32-bit destination register.

### Checklist
- [x] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tt_metal/hw/inc/debug/dprint_tensix.h [hunk 1]
```diff
@@ -5,76 +5,171 @@
...
+            case 2: return (row_id & 0xFFF3) | 0xC;
+            case 3:
+            default: return (row_id & 0xFFF3) | 0x4;
+        }
+    } else {
+        return (row_id & 0xFFF3) | ((row_id & 0x4) << 1) | ((row_id & 0x8) >> 1);
+    }
+}
+
+// calculates dest row address based on
```
Comment (tt-dma) [inline tt_metal/hw/inc/debug/dprint_tensix.h:66]: Incomplete comment?

File: tt_metal/hw/inc/debug/dprint_tensix.h [hunk 2]
```diff
@@ -5,76 +5,171 @@
...
-        if constexpr (print_by_face) {
-            DPRINT << ENDL();
-        }
-
-        for (int row = 0; row < face_r_dim; row++) {
-            // face 3
-            dbg_read_dest_acc_row(row + 3*face_r_dim + 64 * tile_id, rd_data);
-            DPRINT << SETW(6) << " " << TYPED_U32_ARRAY(TypedU32_ARRAY_Format_Tensix_Config_Register_Data_Format_Type, data_format_reg_field_value, rd_data, 8) << ENDL();
+        for (int face_id = 0; face_id < 4; ++face_id) {
+            for (int row_id = 0; row_id < 16; ++row_id) {
```
Comment (tt-dma) [inline tt_metal/hw/inc/debug/dprint_tensix.h:162]: Are the constants 4, 16 just part of our architecture? Could we have defines/constexprs for these?

File: tests/tt_metal/tt_metal/unit_tests_common/dprint/test_print_tensix_dest.cpp
Comment (tt-dma) [inline tests/tt_metal/tt_metal/unit_tests_common/dprint/test_print_tensix_dest.cpp:1]: Thanks for adding a test for this!

File: tests/tt_metal/test_utils/df/float32.hpp [hunk 3]
```diff
@@ -20,13 +20,13 @@ class float32 {
...
 
    public:
-    static const size_t SIZEOF = 4;
+    static constexpr size_t SIZEOF = 4;
 
     // create from float: no rounding, just truncate
     float32(float float_num) {
         static_assert(sizeof float_num == sizeof uint32_data, "Can only support 32bit fp");
         // just move upper 16 to lower 16 (truncate)
-        uint32_data = (*reinterpret_cast<uint32_t*>(&float_num) >> 16);
```
Comment (tt-dma) [inline tests/tt_metal/test_utils/df/float32.hpp:29]: A little confused here, was this a bug in the original code? It doesn't seem to match the comment above, or the `<< 16` in `to_float()`

**Review by tt-dma — APPROVED** [review]

File: tt_metal/hw/inc/debug/dprint_tensix.h (hunk 1 above)
Comment (miacim) [reply tt_metal/hw/inc/debug/dprint_tensix.h]: Fixed. Thank you.

File: tests/tt_metal/test_utils/df/float32.hpp (hunk 3 above)
Comment (miacim) [reply tests/tt_metal/test_utils/df/float32.hpp]: Original code has not been used previously, and I expect that it was not bug, but effectively reduced precision from float32 to float16. I reverted it back to previous version.

File: tt_metal/hw/inc/debug/dprint_tensix.h (hunk 2 above)
Comment (miacim) [reply tt_metal/hw/inc/debug/dprint_tensix.h]: Added constants.

**Pull request merged**